webhook/aws/json_helper.go

```diff
@@ -0,0 +1,57 @@
+package aws
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"fmt"
+)
+
+const (
+	MEDIATYPE_JSON = "application/json"
+)
+
+var (
+	ErrJsonEmpty   = errors.New("JSON payload is empty")
+)
+
+func DecodeJsonPayload(req *http.Request, v interface{}) ([]byte, error) {
+	if req.Body == nil {
+		return nil, ErrJsonEmpty
+	}
+
+	payload, err := ioutil.ReadAll(req.Body)
+	req.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+	if len(payload) == 0 {
+		return nil, ErrJsonEmpty
+	}
+	err = json.Unmarshal([]byte(payload), v)
```

Comment:
Generally speaking, you're exactly right.

However, in this particular case, `DecodeJsonPayload` is being called with a `*SNSMessage` argument for the `v` parameter.  That means the `encoding/json` package will emit numbers as whatever the struct member types are.